File: INTRODUCCIÓN_A_LA_PROGRAMACIÓN_ORIENTADA_A_OBJETOS/ejercicio3/Java/Producto.java
package INTRODUCCIÓN_A_LA_PROGRAMACIÓN_ORIENTADA_A_OBJETOS.ejercicio3.Java;

public class Producto {
    public String nombre;
    public double precio;
    public int stock;

    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    public void vender(int cantidad) {
        if (cantidad <= this.stock) {
            this.stock -= cantidad;
            double totalVenta = cantidad * this.precio;
            System.out.println("Se han vendido " + cantidad + " unidades de " + this.nombre +
                    ". Total: Bs. " + String.format("%.2f", totalVenta));
        } else {
            System.out.println("No hay suficiente stock de " + this.nombre +
                    ". Stock disponible: " + this.stock);
        }
    }

    public void reabastecer(int cantidad) {
        this.stock += cantidad;
        System.out.println("Se han reabastecido " + cantidad + " unidades de " + this.nombre +
                ". Stock actual: " + this.stock);
    }

    public static void main(String[] args) {
        Producto producto = new Producto("Camiseta", 25.0, 100);

        producto.vender(20);
        producto.reabastecer(50);
        producto.vender(150);
    }
}

File: INTRODUCCIÓN_A_LA_PROGRAMACIÓN_ORIENTADA_A_OBJETOS/ejercicio3/Producto.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if cantidad <= self.stock:
            self.stock -= cantidad
            total_venta = cantidad * self.precio
            print(f"Se han vendido {cantidad} unidades de {self.nombre}. Total: Bs. {total_venta:.2f}")
        else:
            print(f"No hay suficiente stock de {self.nombre}. Stock disponible: {self.stock}")

    def reabastecer(self, cantidad):
        self.stock += cantidad
        print(f"Se han reabastecido {cantidad} unidades de {self.nombre}. Stock actual: {self.stock}")
    
# Crear una instancia del producto
producto = Producto(nombre="Camiseta", precio=25.0, stock=100)
producto.vender(20)
producto.reabastecer(50)
producto.vender(150)